### PR TITLE
Added support for specifying a working directory.

### PR DESCRIPTION
**Pull request by markdorison**
* Added support for specifying a working directory if dnsconfig.js is not located in the root of the repository.
* Added examples to README.md.

### CODE REVIEW

File: entrypoint.sh [hunk 1]
```diff
@@ -2,6 +2,8 @@
 
 set -eo pipefail
 
+cd "$INPUT_WORKING_DIR"
+
 IFS=
 OUTPUT="$(dnscontrol "$1")"
```
Comment (koenrh) [inline entrypoint.sh:8]:
```suggestion
OUTPUT="$(dnscontrol "$1" --config "$DNSCONTROL_CONFIG_FILE")"
```

**Review by koenrh — COMMENTED** [review]
Good suggestion. Thanks, @markdorison.

Instead of `cd`-ing into a working directory, what do you think of passing the DNSControl config file location directly to `dnscontrol`? (Defaulting to `dnsconfig.js` in the current directory)

File: entrypoint.sh (hunk 1 above)
Comment (markdorison) [reply entrypoint.sh:8]: @koenrh I am for this, but I believe we will also need the ability to pass in the path to the creds file with `--creds` as well. Does that track with what you are thinking?

File: entrypoint.sh [hunk 2]
```diff
@@ -3,7 +3,7 @@
 set -eo pipefail
 
 IFS=
-OUTPUT="$(dnscontrol "$1")"
+OUTPUT="$(dnscontrol "$1" --config "$INPUT_DNSCONTROL_CONFIG_FILE" --creds "$INPUT_DNSCONTROL_CREDS_FILE")"
```
Comment (markdorison) [inline entrypoint.sh:6]: @koenrh This format is working for me locally but is failing when I try to test this branch in Actions. Do you see anything I am missing?

image

**Review by szczot3k — CHANGES_REQUESTED** [review]
It WILL break if your `dnsconfig.js` is using `require` for anything. You still need to change working directory, beacuse the path in `require` is relative to `dnsconfig.js`.

image

**Comment by koenrh** [comment]
@szczot3k Well spotted! I still like having the ability to explicitly specify paths to the 'config' and 'credentials' file.

Since the `require` functions specify paths relative to the 'config' file, we could figure out the 'config' file directory and change the working directory to that one. That way the `require` functions keep working and you could still specify the paths to the 'config' and 'credentials' file.

```bash
WORKING_DIR="${INPUT_DNSCONTROL_CONFIG_FILE%/*}"
cd "$WORKING_DIR"
```

**Review by szczot3k — APPROVED** [review]
Yep, that works now, even if the `dnsconfig.js` uses `require` relative to config file.

**Comment by markdorison** [comment]
> That way the require functions keep working and you could still specify the paths to the 'config' and 'credentials' file.

I pushed up a change along these lines but it is not yet working as expected. For example, a user specifies the full path to both the config file and the creds file and we then take the directory path to the config file and `cd` into it. We also need to pass the filename of the config file in case it is different from the default (the user probably expects this since they specified it). We are OK to this point (I think), but what if they have also specified a custom path to the creds file? We would have to determine if it is at the same path as the config file since if it is, we have already changed the directory to that path, and if it is at a different path we are in a different relative location then from where the user-specified. Am I over-thinking this?

**Comment by szczot3k** [comment]
Well, tbh I think that you're over-thinking it.

**Comment by koenrh** [comment]
My apologies for the delay!

These changes look good to me, but they didn't work for existing configurations (where the config and credentials arguments where not specified). As I couldn't figure out how to push to this branch, I created a new one: https://github.com/koenrh/dnscontrol-action/pull/41 (including your commits, @markdorison), including a few minor changes. I think this should do. Let me know what you think.

**Comment by koenrh** [comment]
Thank you for your help, @markdorison! (and also @szczot3k) I have merged https://github.com/koenrh/dnscontrol-action/pull/41, which include your changes.

**Pull request closed without merging**